### DISCUSSION TITLE
image registration 3D

### DISCUSSION THREAD

**Discussion by Marie0704** [Project-MONAI/tutorials#1829] [Q&A]
I'm having some trouble adjusting the hyperparameters on another dataset (liver images). Is there a tutorial available that shows how to create a working model? Which parameters are most important and what are the safe ranges to change them?

**Comment by KumoLiu** [comment]
Hi @Marie0704, we have some tutorials focus on the 3d registration, you can refer more details here:
https://github.com/Project-MONAI/tutorials/tree/main/3d_registration

**Reply by Marie0704** [reply]
I know these codes but I haven't found any help with them that would help adapt the model to another dataset

**Reply by kvttt** [reply]
Hi @Marie0704, I'm not sure if this is still relevant. But I think the tutorial [https://github.com/Project-MONAI/tutorials/tree/main/3d_registration](https://github.com/Project-MONAI/tutorials/tree/main/3d_registration) that @KumoLiu pointed out is definitely a good starting point. 

There are a few design choices that need to be made. For example, what image dissimilarity loss to use, what regularization to use, and how to balance between the two loss terms.
1. In terms of image dissimilarity loss, from what you described, I believe MSE suffices because I'm guessing the images that you are trying to register are of the same modality. In addition, MSE is a fairly straightforward loss to compute so it doesn't hurt to try that as a starting point. If MSE doesn't work, try these: [https://monai.readthedocs.io/en/latest/losses.html#registration-losses](https://monai.readthedocs.io/en/latest/losses.html#registration-losses).
2. In terms of the regularization to use for encouraging smooth deformations. You might want to take a look at this tutorial: [https://github.com/Project-MONAI/tutorials/blob/main/modules/bending_energy_diffusion_loss_notes.ipynb](https://github.com/Project-MONAI/tutorials/blob/main/modules/bending_energy_diffusion_loss_notes.ipynb). This tutorial discusses the difference between bending energy and diffusion losses. One rule of thumb that I would like to remember is that: **diffusion loss penalize affine transformation but bending energy does not**. If you believe there are potentially affine transformation between images, use bending energy; otherwise, use diffusion loss. But don't quote me on this. The best regularization to use really depends on what biophysical model best describes the motion of the organs in the images. On this topic, you might find this TMI paper interesting to read: [https://doi.org/10.1109/TMI.2013.2274777](https://doi.org/10.1109/TMI.2013.2274777).
3. In terms of balancing the two loss terms, my suggestion is to try different weights for the regularization term and pick the one that works the best for your use case.

> Which parameters are most important and what are the safe ranges to change them?

I think it is important to choose an appropriate regularization and also an appropriate weight for it. I would start with a weight that balance the image dissimilarity loss and the regularization term such that they are on the same order of magnitude.
